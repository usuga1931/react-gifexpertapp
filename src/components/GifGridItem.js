import React from 'react';

export const GifGrifItem = ({ title, url }) => {

    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    );
}