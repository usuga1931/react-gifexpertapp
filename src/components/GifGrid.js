import React from 'react';
import { useFetchGif } from '../hooks/useFetchGifs';
import { GifGrifItem } from '../components/GifGridItem';

export const GifGrid = ({ category }) => {

    const { data:images, loading } = useFetchGif(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {loading && <p>Cargando</p>}
            <div className="card-grid animate__animated animate__flash">
                {
                    images.map(img => (
                        <GifGrifItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}
