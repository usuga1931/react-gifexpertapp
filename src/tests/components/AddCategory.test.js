import React from 'react';
import { AddCategory } from '../../components/AddCategory';
import { shallow } from 'enzyme';

describe('Pruebas con el componente AddCategory', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('Debe de mostrar correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de cambiar la caja de texto', () => {

        const input = wrapper.find('input');
        const value = 'Hola Mundo';

        input.simulate('change', { target: { value } });

        expect(wrapper.find('p').text().trim()).toBe(value);

    });

    test('No debe postear la informaciÃ³n con submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault() { } });

        expect(setCategories).not.toHaveBeenCalled();

    });

    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {

        const value = 'Homero';
        wrapper.find('input').simulate('change', { target: { value } });

        wrapper.find('form').simulate('submit', { preventDefault() { } });

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        expect(wrapper.find('input').prop('value')).toBe('');

    });
});
