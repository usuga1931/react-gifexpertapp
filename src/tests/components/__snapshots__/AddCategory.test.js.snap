// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente AddCategory Debe de mostrar correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
